src/DTO/Flag.php

```diff
@@ -21,4 +21,86 @@ public function __construct(
         public int $totalShards
     ) {
     }
+
+    public static function fromJson(array $json): Flag
```

Comment:
It's interesting to co-locate the parsing within the class itself. I do kind of like it!